src/Log4Mongo/MongoDBAppender.cs

```diff
@@ -100,23 +100,18 @@ private string GetConnectionString()
 			return connectionStringSetting != null ? connectionStringSetting.ConnectionString : ConnectionString;
 		}
 
-		private MongoDatabase GetDatabase()
+		private IMongoDatabase GetDatabase()
 		{
 			string connStr = GetConnectionString();
 
 			if (string.IsNullOrWhiteSpace(connStr))
 			{
-				return BackwardCompatibility.GetDatabase(this);
+				throw new InvalidOperationException("Must provide a valid connection string");

```

Comment:
In log4net a failing appender does not propagate to the caller, this is by design. This approach is rather heavy because the effect is that the log stop from being sent to Mongo. It would be ok to introduce this sort of changes in a major release, and maybe this is the right moment to do so, but in this case i would just get rid of the if and the `InvalidOperationException`